libs/utils/android/gfxinfo.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2017, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Parser for dumpsys gfxinfo output
+
+import ast, re
+
+def get_value(token):
+    try:
+        v = ast.literal_eval(token)
+    except:
+        return token
+    return v
+
+class GfxInfo(object):
+    """
+    Class for parsing and accessing GfxInfo output
+    """
+    __properties = {}
+
+    def __init__(self, path):
+        """
+        Initialize gfxinfo parser object
+
+        :param path: Path to file containing output of gfxinfo
+        """
+        self.path = path
+        self.parse_gfxinfo()
+
+    def parse_gfxinfo(self):
+        """
+        Parser for gfxinfo output, creates a name/value pair
+        """
+        with open(self.path) as f:
+            content = f.readlines()
+
+        # gfxinfo has several statistics of the format
+        # <string>: <value>
+        for line in content:
+            line = line.rstrip()
+            # Ignore any lines that aren't of this format
+            if not ':' in line:
+                continue
+
+            tokens = line.split(':')
+            # Ignore any lines with ':' but no <value>
+            if (len(tokens[1]) < 1):
```

Comment:
Looks like we could squash this into the if above and have `if ':' not in line or line.endswith(':')`